src/main/java/org/elasticsearch/index/termvectors/ShardTermVectorsService.java

```diff
@@ -332,6 +347,10 @@ private boolean useDfs(TermVectorsRequest request) {
         return request.dfs() && (request.fieldStatistics() || request.termStatistics());
     }
 
+    private boolean useTermVectorsFilter(TermVectorsRequest request) {
+        return request.filterSettings() != null && request.fieldStatistics() && request.termStatistics();
```

Comment:
If filter settings are set but statistics are not requested then we ignore the filter settings?